### PR TITLE
SparkPost: Add template-based message support and basic sending-domain registration functions

### PR DESCRIPTION
**Pull request by nickvanderveeken**

### CODE REVIEW

File: src/Mail/Message/Converter/SparkPostConverter.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace SlmMail\Mail\Message\Converter;
+
+use SlmMail\Mail\Message\SparkPost;
+use SlmMail\Mail\Message\Mandrill;
+
+class SparkPostConverter
```
Comment (roelvanduijnhoven) [inline src/Mail/Message/Converter/SparkPostConverter.php:8]: @nickvanderveeken I would place this code inside our _own_ repo TBH!

File: src/Service/SparkPostService.php [hunk 2]
```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
...
-     * @param  Message $message
-     *
-     * @return string
      */
     protected function prepareBody(Message $message): string
     {
         return $message->getBodyText();
     }
 
+    public function registerSendingDomain(string $domain): bool
```
Comment (roelvanduijnhoven) [inline src/Service/SparkPostService.php:317]: @nickvanderveeken I think it makes more sense to simply pass the DKIM parameter as (optional) arguments to this method. I think in general the DKIM settings belong to this method call; and there is no need for a _default_ DKIM config. 

I could be missing something here; so please let me know if that is the case!

File: src/Service/SparkPostService.php [hunk 3]
```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
...
+        $this->parseResponse($response, [404]);
+    }
+
     /**
      * @param string $uri
      * @param array  $parameters
      * @return HttpClient
      */
-    private function prepareHttpClient(string $uri, array $parameters = []): HttpClient
+    protected function prepareHttpClient(string $uri, array $parameters = []): HttpClient
```
Comment (roelvanduijnhoven) [inline src/Service/SparkPostService.php:382]: Is there a use case for overriding this? I would think that _if_ you simply inject another client to the constructor of this service; to augment or extend the client itself.

**Review by roelvanduijnhoven — COMMENTED** [review]
Thanks for the contributon :). Nice to see the Sparkpost grow in it's supported features!

Some remarks on how I think we can improve DKIM.

Let me know once I can merge this thing; and release a new version.

File: src/Service/SparkPostService.php (hunk 2 above)
Comment (nickvanderveeken) [reply src/Service/SparkPostService.php]: Yeah, it makes more sense.

**Comment by nickvanderveeken** [comment]
@roelvanduijnhoven I addressed your comments, adjusted the PR and succesfully integration-tested it with my application code. I'd say it's good to merge it now.

**Comment by roelvanduijnhoven** [comment]
@nickvanderveeken Launched version v4.1 :). Thanks a lot!

**Pull request merged**